src/overlays/actors/ovl_En_Bombal/z_en_bombal.c

```diff
@@ -32,37 +37,240 @@ const ActorInit En_Bombal_InitVars = {
+    this->actor.targetMode = 6;
+    this->actor.colChkInfo.health = 1;
+    this->scale = 0.1f;
+    this->cutscene = this->actor.cutscene;
+    func_80C05B24(this);
+}
 
-extern UNK_TYPE D_06000A00;
+void EnBombal_Destroy(Actor* thisx, PlayState* play) {
+    EnBombal* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/EnBombal_Init.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/EnBombal_Destroy.s")
+void func_80C05B24(EnBombal* this) {
+    this->isPopped = false;
+    this->actionFunc = func_80C05B3C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/func_80C05B24.s")
+void func_80C05B3C(EnBombal* this, PlayState* play) {
+    Player* player;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/func_80C05B3C.s")
+    this->oscillationAngle += 1500.0f;
+    this->actor.velocity.y = Math_SinS(this->oscillationAngle);
+    Math_ApproachF(&this->scale, 0.1f, 0.3f, 0.01f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/func_80C05C44.s")
+    if (play->msgCtx.msgLength == 0) {
+        if (this->collider.base.acFlags & AC_HIT) {
+            player = GET_PLAYER(play);
+            this->collider.base.acFlags &= ~AC_HIT;
+            if (!(gSaveContext.save.weekEventReg[75] & 0x40) && !(gSaveContext.save.weekEventReg[73] & 0x10) &&
+                !(gSaveContext.save.weekEventReg[85] & 2)) {
+                player->stateFlags1 |= ACTOR_FLAG_20;
+                this->actor.flags |= ACTOR_FLAG_100000;
+            }
+            this->actionFunc = func_80C05C44;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/func_80C05DE8.s")
+void func_80C05C44(EnBombal* this, PlayState* play) {
+    s32 phi_s0 = false;
+    s32 i;
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/EnBombal_Update.s")
+    if (!(gSaveContext.save.weekEventReg[75] & 0x40) && !(gSaveContext.save.weekEventReg[73] & 0x10) &&
+        !(gSaveContext.save.weekEventReg[85] & 2)) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+            return;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/EnBombal_Draw.s")
+        if (!ActorCutscene_GetCanPlayNext(this->cutscene)) {
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+        } else {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutscene, &this->actor);
+            phi_s0 = true;
+        }
+    } else {
+        phi_s0 = true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bombal/func_80C05F90.s")
+    if (phi_s0) {
+        Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+        pos.y += 60.0f;
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, pos.x, pos.y, pos.z, 0xFF, 0xFF, 0xC8,
```

Comment:
In this case the rots seem to be used for colours, so I'd put them in dec (this is obviously going to need a review later, but might as well do it now).